packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -324,14 +325,26 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
         AuthErrorCode.TENANT_ID_MISMATCH
       );
     }
+    await this._runBeforeStateCallbacks(user);
```

Comment:
`_updateCurrentUser` is also called by the storage onevent callbacks, which means a persistence change in another tab could trigger a stop here and the tabs would get out of sync. I'm not sure what the right behavior should be

Reply:
We discussed this, and we want the callbacks to not run if called by _onStorageEvent(), so I made use of the optional `skipBeforeStateCallbacks` argument I added to _updateCurrentUser when called by _onStorageEvent().